Cone_spanners_2/include/CGAL/Cone_spanners_2/Plane_scan_tree.h

```diff
@@ -182,14 +182,14 @@ class Plane_scan_tree {
 
     /* End Iterator */
     inline iterator end() {
-        static iterator res;
+        static const iterator res;
```

Comment:
Note that the *initialization* of `static const ` local variables is not guaranteed to be thread-safe: two different threads could use this function for the first time at the same time. See this [SO answer]. Its only in C++11 and later that the C++ norm adds that guarantee.

[SO answer]: http://stackoverflow.com/a/3237024/1728537

It seems that, for Plain-Old-Data types, the initialization is actually done at compile time by compiler, so is should be safe.

**But:** the type `iterator` (a typedef for `_Iterator`) is actually not a POD for C++<11: it has a user-defined destructor.

In that case, the default-constructor of `_iterator` is so trivial: 
```C++
    _Iterator(leaf_type* start = NULL) : cell (0), leaf(start) {}`
```
that I think it would actually be faster not to use a static variable:
```C++
inline iterator end() const {
  return iterator();
}
inline const_iterator end() const {
  return const_iterator();
}
```